_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+kubectl apply -f store-${suffix}/
+...
+```
+
+You can check on your pods with the following command. Everything was up and running almost immediately for me on AKS.
+
+```bash
+kubectl get pods -n demo
+```
+
+```bash
+NAME                                  READY   STATUS    RESTARTS   AGE
+blog-6896f6dd58-mbjkn                 1/1     Running   0          3m51s
+blog-neo4j-0                          1/1     Running   0          3m48s
+gateway-7f6d57765f-2fhfb              1/1     Running   0          3m46s
+gateway-postgresql-647476b4d5-jdp5c   1/1     Running   0          3m44s
+jhipster-registry-0                   1/1     Running   0          3m52s
+jhipster-registry-1                   1/1     Running   0          3m39s
+store-7889695569-k4wkv                1/1     Running   0          3m41s
+```
+
+Here's an example of what this looks like with Kubernetes Lens (the containers were still booting).
+
+{% img blog/jhipster-k8s-azure/lens-booting.png alt:"K8s Lens screenshot" width:"1000" %}{: .center-image }
+
+Another useful command is `describe`. I won't replicate the output here, but if you want more detailed information for debugging, you can also run the following.
+
+```bash
+kubectl describe pods -n demo
+```
+
+To tail logs, you can use the name of the pod.
+
+```
+kubectl logs <pod-name> --tail=-1 -n demo
+```
+
+Although, like I said, k9s and Lens are really the way to go for more detailed inspection.
+
+Here's a screenshot from k9s. You can dig down into the different pods to get more detailed information and inspect logs.
+
+{% img blog/jhipster-k8s-azure/k9s-screenshot.png alt:"Authenticated" width:"1000" %}{: .center-image }
+
+You can use port-forwarding to see the JHipster Registry.
+
+```
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Open a browser and navigate to `http://localhost:8761`. You will be redirected to the Okta login screen, after which you will be taken to the registry.
+
+Make sure everything is green. If you have an error, check the logs for the pod that caused the error. You can restart a specific deployment by deleting it and re-applying it. For example, to restart the store, you can use the commands below from the `k8s` directory. I found that I sometimes had to delete and restart the store to get it to work.
+
+```bash
+kubectl delete -f store-k8s/
+kubectl apply -f store-k8s/
+```
+
+Once all is good, `control-c` to stop forwarding the registry. Expose the gateway and open it. 
+
+```
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Go to `http://localhost:8080`. 
+
+Authenticate with Okta. Make sure you can add blogs, posts, tags, and products. Because the store service uses the same Cosmos DB instance that you were using with Docker Compose locally, any test products you created earlier will still be there.
+
+In preparation for the next step. delete everything you just deployed to AKS in the `demo` namespace.
+
+```bash
+kubectl delete all --all -n demo
```

Comment:
This seems to have worked but I have to admin that `all --all` looks kinda funny.